### DISCUSSION TITLE
TSML Updated : now extremely slow

### DISCUSSION THREAD

**Discussion by jcookson-infoblox** [code4recovery/12-step-meeting-list#418] [Q&A] [answered]
**Describe the bug**
After updating to 3.11.0, my area's meetings portion seems extremely slow.  Updates and imports are taking upwards of 2-3x longer.  Am getting frequent 503's on updates.  Was there some backend changes done with this update? 
Wondering if there are any other reported performance issues after upgrading to 3.11.0.

**To Reproduce**
1. Update TSML from previous version, say most recent WP 5.7 compatible version prior to this.
2. Select meetings section
3. Import meetings -or-
4. Update meetings

**Expected behavior**
Some delay is expected for processing changes and imports, but anything longer than say 5-10 secs should be looked into.  And, 503's are not kosher.

**Laptop/Desktop (please complete the following information):**
 - OS: [iOS]
 - Browser [chrome]
 - Version [Version 91.0.4472.114]

**Comment by jcookson-infoblox** [comment]
Additional data, I sometimes open multiple browser tabs with different meeting info in each one to work en-tandem.  Have encountered 503's (a lot and predictably regularly) when submitting more than 2 diff tabs (each with diff meeting info submitted one after the other).  This should be allowed, because each tab should be placing a lock on that specific meeting's info and none others.  So submissions en-tandem would be processed one after the other.

**Comment by tech2serve** [comment]
Please clear browser and website cache. And try again. This is a total guess, but it is the best I have with the information provided :-)

I’m really not sure what is going on, but our update didn’t touch import logic.

In the past we’ve had people experience import issues when the hosting providing made changes. You might also check with them.

What PHP version are you using? And errors showing in the console (outside of the 503s)? 

If you close other tabs, does the import speed up?

Are all imports slow? Or only from some districts? 

Are you importing CSVs or all JSON feeds from other sites? 

What version of the plugin are they using?

I’m converting this to a Discussion which is where we are trying to move public conversation to. If we confirm a bug, I’ll open a new issue.

**Comment by jcookson-infoblox** [comment, marked as answer]
It was most prevalent when editing meetings.  All my imports are via JSON feeds currently.  I have cleared the site cache as well.  I'll see about updating the PHP version and see if that helps.  When I close editing tabs it does not speed up, which leads me to think its on the records lock.

**Reply by tech2serve** [reply]
One user [reported disabling Cloudfare](https://github.com/code4recovery/12-step-meeting-list/issues/311) cleared up his slow import issue.

I will add that when I saw this with another site but doing CSV import, we obtained a copy of the CSV file and imported it without a hitch on our dev sites.

**Reply by tech2serve** [reply]
So @jcookson-infoblox Is this problem resolved?

**Reply by jcookson-infoblox** [reply]
ok, firstly ... most of the slowness is on the **editing** a meeting side.  have tried to utilize multiple browser widows to potentially get more perf out of the app, and have gotten 504's.instead.  presently, most recent edit attempts received 504's - so not sure where to proceed.  second, site is scheduled for PHP upgrade today, will advise if the perf gets better on that front.

**Reply by tech2serve** [reply]
Thanks for the update. 

We've had over 51% adoption of 3.11, and this is the only reported case of slowness. I won't rule out anything at this stage, but I'm leaning towards there being something with your site. 

As I said above, please research and report back things like CDN and server side cache settings and clean up. Also look at disabling other plugins to rule out interference.

And keep monitoring the dev console for any errors.

**Comment by digimech** [comment]
Lee from Greater Seattle Intergroup here - we've noticed it being slow here as well. Just updated to PHP 7.4.21 on SiteGround, clean tech, new cache. Updating the imports is taking much longer, and office team is reporting that just updating meetings is now a struggle. Is there some new overhead?

**Reply by Grateful-Viv** [reply]
I am on SiteGround too, running PHP 7.4.21 under WP 5.8.

Using Firefox Developer edition with Windows 10. I had issues with Chrome and that is why I switched browsers.

Anyway, just did an import of 130 files and it was under 15 seconds. And updates to meetings are instantaneous. 

Perhaps try a different browser?

**Comment by brianw-area24** [comment]
Something I wrote into version 3.11.0 of the plugin caused the plugin to make extra calls to the database, I'm pretty sure this is why you're seeing performance problems. The update we're currently working on should correct this problem. This update is still being worked on and tested.
